### PR TITLE
feat: Replace the use of `keyboardEventListener` mixin to a composable (Part -3)

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR will replace usage of `keyboardEventListener` mixin with the `useKeyboardEvents` composable. And updated some component from option API to composition API.

**Files included in this PR**
1. dashboard/modules/notes/components/AddNote.vue (Updated to composition API)
2. dashboard/routes/dashboard/conversation/labels/LabelBox.vue
3. dashboard/routes/dashboard/helpcenter/components/ArticleSearch/Header.vue (Updated to composition API)
4. dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue (Removed the mixin and use it in **Header.vue**)
5. shared/components/ui/dropdown/DropdownMenu.vue (Updated to composition API)
6. superadmin_pages/views/playground/Index.vue 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**N.B:** Please verify the all other existing functionality of the component that has updated from option API to composition API.

### Allowed on input focus

**AddNote.vue**
1.`$mod+Enter` ⇒ Add notes

**LabelBox.vue**
2. `Escape` ⇒ Close the label dropdown menu in the conversation page sidebar.

**Header.vue**
3. `Escape` ⇒ Close the insert article pop over from the chat screen

**DropdownMenu.vue**
4. `ArrowUp` ⇒ Focus the previous item in the dropdown menu
5.`ArrowDown` ⇒ Focus the next item in the dropdown menu

**playground/Index.vue** (Super admin playground)
5. `Ctrl/Cmd + Enter` ⇒ Send message 

### Not allowed in input focus

**LabelBox.vue**
1.`KeyL` ⇒ Open the label dropdown menu in the conversation page sidebar.

**Header.vue**
2. `Slash` ⇒ Focus the search input in the insert article pop over from the chat screen

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by scmmishra — DISMISSED** [review]
~@iamsivin I've added a `TODO` comment with some enhancements. Since the scope of the PR is just migration, we can leave it as it is and take it up in the future. For now this looks good to me. Approving!~

Ignore accidentally posted on the wrong PR

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/Header.vue [hunk 1]
```diff
@@ -1,53 +1,60 @@
...
-    onBlur() {
-      this.isInputFocused = false;
+});
+
+const emit = defineEmits(['search', 'close']);
+
+const articleSearchHeaderRef = ref(null);
+const searchInputRef = ref(null);
+const searchQuery = ref('');
+const isInputFocused = ref(false);
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/Header.vue:17]: Are we using the `isInputFocused` anywhere?

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue:1]: Was this intentionally removed?

**Review by scmmishra — COMMENTED** [review]
A few comments, other than that LGTM!

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/Header.vue (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/Header.vue]: @scmmishra Yeah, we are not using removed it. https://github.com/chatwoot/chatwoot/pull/9897/commits/3436f846b58048be3e76eb2ef96865182332f4ba

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue:1]: @scmmishra Yes, moved the keyboard events from here and used it in `Header.vue` component. https://github.com/chatwoot/chatwoot/pull/9897/files#diff-3208c9bb986b5789c1729d56efed917b871af1d52af0a3aaa49a38ea388921c6R36

**Comment by iamsivin** [comment]
> A few comments, other than that LGTM!

@scmmishra Updated.

**Review by scmmishra — APPROVED** [review]

**Pull request merged**